src/validation.c

```diff
@@ -407,7 +407,14 @@ validate_netdef_grammar(const NetplanParser* npp, NetplanNetDefinition* nd, GErr
         // LCOV_EXCL_STOP
     }
 
-    if (nd->type == NETPLAN_DEF_TYPE_NM && (!nd->backend_settings.passthrough || !g_datalist_get_data(&nd->backend_settings.passthrough, "connection.type")))
+    GHashTable* key = NULL;
+    if (nd->backend_settings.passthrough) {
+        GHashTable* group = g_hash_table_lookup(nd->backend_settings.passthrough, "connection");
+        if (group) {
+            key = g_hash_table_lookup(group, "type");
+        }
+    }
+    if (nd->type == NETPLAN_DEF_TYPE_NM && (!nd->backend_settings.passthrough || !key))
```

Comment:
I don't think we need the `!nd->backend_settings.passthrough` here any more. `key` is only defined if `nd->backend_settings.passthrough` was accessible in the previous if-statement.